README.md

```diff
@@ -501,6 +501,19 @@ The previous command might cause your testing to die along the way due to memory
 $ cargo test --jobs <N>
 ```
 
+### Bench execution
+To run the benchmarks and generate a performance report:
+
+```
+$ cargo bench
+```
+
+This command will compile the benchmarks and run them using all available CPUs on your machine. Criterion will perform multiple iterations of each benchmark to collect performance data and generate statistical analysis.
+
+Check the report created at `target/criterion/report/index.html`
```

Comment:
What do you see as the purpose of this report and the test itself? I.e. how do you think the numbers should be interpreted and potentially addressed? This is only time-related, whereas the original issue asks for memory investigation (which is done in another test, though).

Reply:
As a performance measurement, it should help in making the node faster and prevent performance regressions as you run it after each PR. You are correct in that the original issue asked specifically about memory but I thought using this as a proxy measurement will also help in this issue.